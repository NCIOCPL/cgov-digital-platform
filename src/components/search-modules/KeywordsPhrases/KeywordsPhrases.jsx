import React from 'react';
import { Fieldset, TextInput } from '../../atomic';
import './KeywordsPhrases.scss';

const KeywordsPhrases = ({ handleUpdate }) => {
  return (
    <Fieldset
      id="keyword"
      legend="Keywords/Phrases"
      helpUrl="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#keywords"
    >
      <TextInput
        action={handleUpdate}
        id="q"
        name="keywordPhrases"
        label="Search by word or phrase (use quotation marks with phrases)."
        placeHolder="Examples: PSA, 'Paget disease'"
      />
    </Fieldset>
  );
};

export default KeywordsPhrases;
