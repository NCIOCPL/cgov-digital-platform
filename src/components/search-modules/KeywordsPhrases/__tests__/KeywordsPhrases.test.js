import React from 'react';
import { render, shallow, mount } from 'enzyme';
import {
  resetDom,
  setupDom
} from '../../../../utilities/testHelpers';
import KeywordsPhrases from '../KeywordsPhrases';

const defaultProps = {
}

const setup = (enzymeMethod = render, props = {}) => {
  const component = enzymeMethod(
    <KeywordsPhrases {...props} />
  );
  return component;
};

describe('KeywordsPhrases', () => {
  describe('Render', () => {
    let component = null;

    afterEach(() => {
      resetDom();
    });

    it('renders without error', () => {
      component = setup(render, defaultProps);
      expect(component).toMatchSnapshot();
    });
  });

});
