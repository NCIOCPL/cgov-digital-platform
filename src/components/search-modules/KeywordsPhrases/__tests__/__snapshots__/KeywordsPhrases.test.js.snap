// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeywordsPhrases Render renders without error 1`] = `
<fieldset
  class="cts-fieldset "
  id="fieldset--keyword"
>
  <legend
    class="cts-fieldset__legend"
  >
    <span>
      Keywords/Phrases
    </span>
  </legend>
  <a
    aria-label="Help"
    class="text-icon-help"
    href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#keywords"
    rel="noopener noreferrer"
    target="_blank"
  >
    ?
  </a>
  <div
    class="cts-fieldset__body"
  >
    <div
      class=""
    >
      <label
        class="cts-label"
        for="q"
        id="q-label"
      >
        Search by word or phrase (use quotation marks with phrases).
      </label>
      <input
        aria-labelledby="q-label"
        aria-required="false"
        class="cts-input null"
        id="q"
        name="keywordPhrases"
        placeholder="Examples: PSA, 'Paget disease'"
        spellcheck="false"
        type="text"
        value=""
      />
    </div>
  </div>
</fieldset>
`;
